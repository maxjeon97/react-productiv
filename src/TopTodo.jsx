import React from "react";
import Todo from "./Todo";
import "./TopTodo.css";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo -> Todo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  const uncompletedTodos = todos.filter(todo => todo.isDone === false);

  let top = uncompletedTodos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, uncompletedTodos[0]);

  return (
    <div className="TopTodo">
      <Todo todo={top} />
    </div>
  );
}

export default TopTodo;