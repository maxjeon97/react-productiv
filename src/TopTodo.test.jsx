import { describe, test, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const testTodos = [
    {
        id: 1,
        title: "Test1",
        description: "Test1 Descr",
        priority: 1,
        isDone: false,
    },
    {
        id: 2,
        title: "Test2",
        description: "Test2 Descr",
        priority: 2,
        isDone: false,
    },
    {
        id: 3,
        title: "Test3",
        description: "Test3 Descr",
        priority: 3,
        isDone: false,
    },
];

describe("TopTodo component", function () {
    test("renders without crashing", function () {
        render(<TopTodo todos={testTodos}/>);
    });

    test('matches snapshot', function () {
        const { container } = render(<TopTodo todos={testTodos}/>);
        expect(container).toMatchSnapshot();
    });

    test("renders the correct todo (highest priority)", function () {
        const result = render(<TopTodo todos={testTodos}/>);
        expect(result.queryByText("Test1")).toBeInTheDocument();
        expect(result.queryByText("Test2")).not.toBeInTheDocument();
        expect(result.queryByText("Test3")).not.toBeInTheDocument();
    });
});