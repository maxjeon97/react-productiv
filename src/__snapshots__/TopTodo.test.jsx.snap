// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopTodo component > matches snapshot 1`] = `
<div>
  <div
    class="TopTodo"
  >
    <div
      class="Todo"
      id="1"
    >
      <div
        class="Todo-title"
      >
        <b>
          Test1
        </b>
        <small>
          (priority: 
          1
          )
        </small>
      </div>
      <div>
        <small>
          Test1 Descr
        </small>
      </div>
    </div>
  </div>
</div>
`;
